flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2011,14 +2006,42 @@ class ArrayExprLowering {
     return x.Rank() != 0;
   }
 
+  // Generate an error message, but no code.
+  CC errorCC(llvm::StringRef msg) {
+    mlir::emitError(getLoc(), msg);
+    return [](IterSpace) { return mlir::Value{}; };
+  }
+
+  CC genProcRef(const Fortran::evaluate::ProcedureRef &procRef,
+                llvm::Optional<mlir::Type> retTy) {
+    auto loc = getLoc();
+    if (procRef.IsElemental()) {
+      if (const auto *intrin = procRef.proc().GetSpecificIntrinsic()) {
+        (void)intrin; // NB: LLVM_ATTRIBUTE_UNUSED doesn't suppress warning
+        TODO(loc, "elemental intrinsic");
+      }
+      if (ScalarExprLowering::isStatementFunctionCall(procRef))
```

Comment:
Are elemental statement function really a thing ? (they may be, I am just surprised).

Reply:
I don't think that statement functions can be elemental procedures (though they can be pure procedures). So it's a bug that we don't catch them in semantics.

Reply:
Thanks, Tim. I'll just make this a hard error then.